### PR TITLE
nix: fix fish shell completions

### LINKED ISSUE #1408: Fish shell completions are not loaded on Nix
## Description

When you install `jj` with `nix profile install ..` the fish completions are not recognized by fish.

This installation does ~~add the file~~ link `~/.nix-profile/share/fish/vendor_completions.d/jujutsu.fish`, and that folder is present in `$fish_complete_path` - so you'd expect it to work, but it looks like fish [requires](https://fishshell.com/docs/current/completions.html#where-to-put-completions) the filename to be `command-name.fish`, which it isn't in our case, as the filename is `jujutsu.fish` and command is `jj`.

So a fix should be to update the [flake](https://github.com/martinvonz/jj/blob/522de4f8c72ffb0cd030613a3796855b5cfcd9e2/flake.nix#L72) to create `jj.fish` specifically

(No idea about other shells)

### PR DESCRIPTION
**Pull request by necauqua**
I've tested it and my guess was right, so pushing the oneline fix
Fixes #1408

### CODE REVIEW

**Review by martinvonz — APPROVED** [review]
Thanks!

File: flake.nix [hunk 1]
```diff
@@ -69,7 +69,7 @@
             $out/bin/jj support completion --bash > ./completions.bash
             installShellCompletion --bash --name ${pname}.bash ./completions.bash
             $out/bin/jj support completion --fish > ./completions.fish
-            installShellCompletion --fish --name ${pname}.fish ./completions.fish
+            installShellCompletion --fish --name jj.fish ./completions.fish
```
Comment (samueltardieu) [inline flake.nix:72]: Why would this a problem for this line only and not for the other lines using `${pname}`?

Comment (necauqua) [reply flake.nix]: I've actually decided to test other shells, and while `zsh` worked, `bash` required similar change, you were right.

In the end, I've looked at how nixpkgs packages do `installShellCompletion` and found a cleaner way.

Comment (samueltardieu) [reply flake.nix]: I prefer this way indeed.

**Pull request merged**